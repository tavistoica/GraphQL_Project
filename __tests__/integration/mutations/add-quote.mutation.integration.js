const { createTestClient } = require("apollo-server-testing");
const { gql } = require("apollo-server");
const {
  database: { url },
} = require("../../../src/core/config/test.config");
const { Server } = require("../../../src/bootstrap/server.bootstrap");
const { Database } = require("../../../src/bootstrap/database.bootstrap");

const server = new Server();
const database = new Database();

function getMutation() {
  return gql`
    mutation AddQuotes {
      addQuote(
        input: {
          phrase: "Two things are infinite: the universe and human stupidity; and I'm not sure about the universe."
          quoter: "Albert Einstein"
        }
      ) {
        phrase
      }
    }
  `;
}

async function createContext() {
  return {
    mutation: getMutation(),
  };
}

beforeEach(async () => {
  await database.bootstrap({ url });
  server.bootstrap();
});
afterEach(async () => {
  await database.reset();
  await server.close();
});

it("adds a quote", async () => {
  const { mutation } = await createContext();
  const { mutate } = createTestClient(server.apolloClient);
  const response = await mutate({ mutation });
  expect(response).toMatchSnapshot();
});
