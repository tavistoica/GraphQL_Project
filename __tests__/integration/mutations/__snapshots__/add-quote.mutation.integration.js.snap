// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adds a quote 1`] = `
Object {
  "data": Object {
    "addQuote": Object {
      "phrase": "Two things are infinite: the universe and human stupidity; and I'm not sure about the universe.",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
